src/main/java/org/gridsuite/explore/server/services/StudyService.java

```diff
@@ -6,138 +6,123 @@
+        String path = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION +
+                        "/studies?studyUuid={studyUuid}")
+                .buildAndExpand(studyUuid)
+                .toUriString();
+        try {
+            multipartBodyBuilder.part("caseFile", caseFile.getBytes()).filename(Objects.requireNonNull(caseFile.getOriginalFilename()));
+        } catch (IOException e) {
+            throw new ExploreException(IMPORT_CASE_FAILED);
+        }
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.MULTIPART_FORM_DATA);
+        headers.add(HEADER_USER_ID, userId);
+        HttpEntity<MultiValueMap<String, HttpEntity<?>>> request = new HttpEntity<>(
+                multipartBodyBuilder.build(), headers);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, request, Void.class);
+        } catch (HttpStatusCodeException e) {
+            throw new ExploreException(INSERT_STUDY_FAILED);
+        }
     }
 
-    public Mono<Void> insertStudy(UUID sourceStudyUuid, UUID studyUuid, String userId) {
+    public void insertStudy(UUID sourceStudyUuid, UUID studyUuid, String userId) {
         String path = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION +
                         "/studies")
                 .queryParam("duplicateFrom", sourceStudyUuid)
                 .queryParam("studyUuid", studyUuid)
                 .toUriString();
-
-        return webClient.post()
-                .uri(studyServerBaseUri + path)
-                .header(HEADER_USER_ID, userId)
-                .retrieve()
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
-
+        restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, new HttpEntity<>(getHeaders(userId)), Void.class);
     }
 
     @Override
-    public Mono<Void> delete(UUID studyUuid, String userId) {
+    public void delete(UUID studyUuid, String userId) {
         String path = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION + "/studies/{studyUuid}")
                 .buildAndExpand(studyUuid)
                 .toUriString();
-
-        return webClient.delete()
-                .uri(studyServerBaseUri + path)
-                .header(HEADER_USER_ID, userId)
-                .retrieve()
-                .onStatus(httpStatus -> httpStatus != HttpStatus.OK, ClientResponse::createException)
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, new HttpEntity<>(getHeaders(userId)), Void.class);
+        } catch (HttpStatusCodeException e) {
+            if (HttpStatus.OK != e.getStatusCode()) {
```

Comment:
this check will always be true, also, use REMOTE_ERROR

Reply:
the behaviour was like this
return webClient.delete()
                .uri(studyServerBaseUri + path)
                .header(HEADER_USER_ID, userId)
                .retrieve()
                .onStatus(httpStatus -> httpStatus != HttpStatus.OK, ClientResponse::createException)
                .bodyToMono(Void.class)
                .publishOn(Schedulers.boundedElastic())
                .log(ROOT_CATEGORY_REACTOR, Level.FINE);

i will change HttpMethod.POST with HttpMethod.DELETE